tests/utils/test_multiprocessing.py

```
@@ -306,3 +307,40 @@ def test_stop_runs_housekeeping_after_interrupt():
+
+
+def test_forking_restores_filters_after_normal_exit():
+    before = warnings.filters[:]
+    with forking():
+        pass
+    assert warnings.filters == before
+
+
+def test_forking_restores_filters_after_exception():
+    before = warnings.filters[:]
+    with pytest.raises(RuntimeError, match="^boom$"):
```

Ruff (0.15.21)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)